.github/workflows/main.yml

```
@@ -0,0 +1,57 @@
+name: CI
+
+on: [push, pull_request]
+
+env:
+  COLORTERM: 'yes'
+  TERM: 'xterm-256color'
+  PYTEST_ADDOPTS: '--color=yes'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - "2.7"
+          - "3.6"
+          - "3.7"
+          - "3.8"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install dependencies
+        run: pip install -r requirements-test.txt
+      - name: Install RPM tools
+        run : |
+          sudo curl --create-dirs -o /usr/local/bin/spectool https://pagure.io/rpmdevtools/raw/26a8abc746fba9c0b32eb899b96c92841a37855a/f/spectool.in
+          sudo curl --create-dirs -o /usr/local/bin/rpmdev-bumpspec https://pagure.io/rpmdevtools/raw/6f387c1deaa5cbed770310e288abde04b17421dc/f/rpmdev-bumpspec
+          sudo curl --create-dirs -o /usr/local/bin/rpmdev-packager https://pagure.io/rpmdevtools/raw/master/f/rpmdev-packager
+          sudo chmod +x /usr/local/bin/*
+          sudo apt-get update && sudo apt-get install -y git-annex rpm
+      - name: Run tests
+        run: tox
+
+  container:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run container tests
+        run: docker build --build-arg REPO=${GITHUB_REPOSITORY} --build-arg VERSION=${GITHUB_REF} -t quay.io/foreman/obal:latest .
```

Nice. I wonder if the tag should be PR, possible with the PR number itself. Right now there's no permission to push, but I think it's good practice to prevent it from even happening.
```suggestion
        run: docker build --build-arg REPO=${GITHUB_REPOSITORY} --build-arg VERSION=${GITHUB_REF} -t quay.io/foreman/obal:pr .
```

Also wondering if you can run the container that was just built and run `obal --help` as a minimal integration test.

---

I couldn't find the PR number in the data we get inside the action. Otherwise I'd prefer to pull `refs/pull/N/merge` as that would have provided the merged result instead of the fork branch (which might be not rebased properly).